### DISCUSSION TITLE
Dynamic base paths by folder

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#42877] [Ideas]
### Describe the feature you'd like to request

Ability to change `basePath` in `next.config.js` from `getStaticProps`.

We ([Mintlify](https://mintlify.com/)) use Next.js as a multi-tenant application. That is, we have multiple customer websites out of a single Next.js instance. Customers have a unique subdomain that a middleware rewrites into folders.

Specifically, we have this structure:

`/_site/[subdomain]/[[...slug]].ts`

The subdomain is filled in by the middleware such that each customer has their own folder.

Some customers want to host their sites out of a subdirectory (eg. `mintlify.com/docs`) others want a subdomain (eg. `docs.mintlify.com`).

We need a way of changing the basePath inside getStaticProps so bundles are places in `/docs/_next/` or `/_next/` depending on each customer's choice. The basePath is the same for each page in the [subdomain] folder.

### Describe the solution you'd like

getStaticProps can include `basePath` in the object returned to change the `basePath` during build-time. The basePath cannot be changed after the site is built with ISR.

### Describe alternatives you've considered

We can manually re-write paths to most files. For example, we tried using `${basePath}${href}` in links and manually adding the base path in getStaticPaths.

This mostly works! You can see it here: https://mintlify.mintlify.app/docs/quickstart

The issue happens when you host the site out of a subdirectory with Cloudflare workers, references to `/_next/` are routed to the apex server. 

We tested hosting mintlify.com/docs by manually added basePath and it worked. The CSS is imported like: `/docs/_next/static/css/ae6af72ecec203a5.css` but without `basePath` we get `/_next/static/css/ae6af72ecec203a5.css`. This proves the basePath functionality in next.js works, but we still need a way of dynamically changing it per subdomain.

**Comment by hahnbeelee** [comment]
Overall, for a multi-tenant application, any setting that is set in the next.config.json is really cumbersome because we want different settings for the different sites. Would it be difficult to move basePath out of the config? We would be willing to take a stab at it if someone could offer insight into whether there's a viable way to approach this?
